mlflow/cli.py

```diff
@@ -566,7 +587,45 @@ def gc(older_than, backend_store_uri, run_ids):
     else:
         run_ids = run_ids.split(",")
 
-    for run_id in run_ids:
+    if not skip_experiments:
+        deleted_older_experiment_ids = []
+        next_experiment_page_token = None
+        time_threshold = get_current_time_millis() - time_delta
+        filter_string = f"last_update_time < {time_threshold}" if older_than else ""
+        while True:
+            page_results = backend_store.search_experiments(
+                view_type=ViewType.DELETED_ONLY,
+                filter_string=filter_string,
+                page_token=next_experiment_page_token,
+            )
+            for experiment in page_results:
+                deleted_older_experiment_ids.append(experiment.experiment_id)
+            if page_results.token is None:
+                break
+            next_experiment_page_token = page_results.token
+
+        if not experiment_ids:
+            experiment_ids = deleted_older_experiment_ids
+        else:
+            experiment_ids = experiment_ids.split(",")
+
+    next_run_page_token = None
+    while True:
+        page_results = backend_store.search_runs(
+            experiment_ids=experiment_ids,
+            filter_string="",
+            run_view_type=ViewType.DELETED_ONLY,
+            page_token=next_run_page_token,
+        )
+
+        for run in page_results:
+            run_ids.append(run.info.run_id)
+
+        if page_results.token is None:
+            break
+        next_run_page_token = page_results.token
```

Comment:
I think this block needs to be indented.

Reply:
I kept this block outside of `if not skip_experiments` to search for runs if experiments are skipped. Do I need to add this block into `if not skip_experiments`?

Reply:
Do you think we should delete the runs even when their experiments are NOT going to be deleted?

Reply:
We don't need to delete them. But, we can delete only those runs(with `ViewType.DELETED_ONLY`) which are user mentioned.

Reply:
> We don't need to delete them.

I think the current implementation deletes them.